packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -171,6 +182,30 @@ contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver, Tra
         return Encoding.encodeVersionedNonce(msgNonce, messageVersion);
     }
 
+    /// @notice Generates a unique hash for a message to be relayed across chains. This hash is
+    ///         used to identify the message and ensure it is not relayed more than once.
+    /// @param _destination Chain ID of the destination chain.
+    /// @param _source Chain ID of the source chain.
+    /// @param _nonce Unique nonce associated with the message to prevent replay attacks.
+    /// @param _sender Address of the user who originally sent the message.
+    /// @param _target Address of the contract or wallet that the message is targeting on the destination chain.
+    /// @param _message The message payload to be relayed to the target on the destination chain.
+    /// @return Hash of the encoded message parameters, used to uniquely identify the message.
+    function _relayMessageHash(
```

Comment:
Usually we have a library that is used for this sort of thing rather than an internal function, having it in a library allows devs to import the library and construct the message hashes themselves. I recommend either adding a function to `Hashing.sol` or just inlining the hash in both places rather than making an internal function